airframe-sql/src/main/scala/wvlet/airframe/sql/analyzer/TypeResolver.scala

```diff
@@ -183,11 +183,11 @@ object TypeResolver extends LogSupport {
 
   def resolveRegularRelation(context: AnalyzerContext): PlanRewriter = {
     case filter @ Filter(child, filterExpr, _) =>
-      filter.transformExpressions { case x: Expression => resolveExpression(context, x, filter.inputAttributes) }
+      filter.transformUpExpressions { case x: Expression => resolveExpression(context, x, filter.inputAttributes) }
```

Comment:
The previous transformExpression implementation was wrongly using depth-first traversal (so changed to breadth-first traversal in this PR). Do transformation in depath-first manner with transformUpExpressions.